File: Assets/CS_Scripts/PlayerCollision.cs
// handles collision interaction between player
// and objects in environment

using UnityEngine;
using System.Collections;

// handles interaction with bunnies and lozenges
// if enough bunnies are collected, go into high score screen
public class PlayerCollision : MonoBehaviour {

	void OnTriggerEnter(Collider collisionInfo) {
		if (collisionInfo.gameObject.tag == "bunny") {
			BunnyCollect.numBunniesCaught++;
			Destroy(collisionInfo.gameObject);
			
			if(BunnyCollect.numBunniesCaught == BunnyCollect.numGoalBunnies){
				WinText.textOn = true;
				WinText.message = "you got all 5 bunnies!  you win!";
				Wait();
				Application.LoadLevel("HighScore");
			}
			
		}else if (collisionInfo.gameObject.tag == "lozenge"){
			LozengeDefeat.numLozengesDefeated++;
			Destroy(collisionInfo.gameObject);
		}
	}
	
	// function for waiting
	// waits 2 seconds
	IEnumerator Wait() {
		yield return new WaitForSeconds(2f);
	}
}

File: Assets/Scripts/PlayerCollision.js
// handles interaction with bunnies and lozenges
// if enough bunnies are collected, go into high score screen

function OnTriggerEnter(collisionInfo : Collider) {
	if (collisionInfo.gameObject.tag == "bunny"){
		BunnyCollect.bunniesCaught++;
		Destroy(collisionInfo.gameObject);
		if(BunnyCollect.bunniesCaught == BunnyCollect.numGoalBunnies){
			WinText.textOn = true;
			WinText.message = "you got all 5 bunnies!  you win!";
			yield new WaitForSeconds(2);
			Application.LoadLevel("HighScore");
		}
		Debug.Log("bunny trigger!");
	}
	
	else if (collisionInfo.gameObject.tag == "lozenge"){
		LozengeDefeat.numLozengesDefeated++;
		Destroy(collisionInfo.gameObject);
		Debug.Log("lozenge was gotten by something");
	}
	
	else Debug.Log("other trigger...");
}

@script RequireComponent(AudioSource)